adapters/repos/db/roaringset/compactor.go

```diff
@@ -22,6 +22,74 @@ import (
 	"github.com/weaviate/weaviate/adapters/repos/db/lsmkv/segmentindex"
 )
 
+type compactorWriter interface {
+	segmentindex.SegmentWriter
+	Reset(io.Writer)
+}
+
+type memoryWriter struct {
+	buffer *[]byte
+	pos    *int
+	maxPos *int
+	writer io.WriteSeeker
+}
+
+// NewMemoryWriterWrapper creates a new memoryWriter with initialized pointers
+func newMemoryWriterWrapper(initialCapacity int64, writer io.WriteSeeker) *memoryWriter {
+	buf := make([]byte, initialCapacity)
+	pos := 0
+	maxPos := 0
+	return &memoryWriter{
+		buffer: &buf,
+		pos:    &pos,
+		maxPos: &maxPos,
```

Comment:
question (non-blocking): Can you explain the motivation behind all the individual pointers for `pos`, `maxPos`, etc? Would it not be sufficient to have a pointer to the `memoryWriter` and access through `m.pos`, etc?

Reply:
yes 😅